rust/src/examples.rs

```diff
@@ -0,0 +1,214 @@
+/*!
+A guide to getting started with the XayNet examples.
+
+# Examples
+
+The XayNet examples code can be found under the `rust/examples` directory of the
+[`xaynet`](https://github.com/xaynetwork/xaynet/) repository.
+
+This Getting Started guide will cover only the general ideas around usage of the
+examples. Also see the source code of the individual examples themselves, which
+have plenty of comments.
+
+Running an example typically requires having a *coordinator* already running,
+which is the core component of XayNet.
+
+# Federated Learning
+
+A federated learning session over XayNet consists of two kinds of parties - a
+*coordinator* and (multiple) *participants*. The two parties engage in a
+protocol (called PET) over a series of rounds. The over-simplified idea is that
+in each round:
+
+1. The coordinator makes available a *global* model, from which selected
+participants will train model updates (or, *local* models) to be sent back to
+the coordinator.
+
+2. As a round progresses, the coordinator aggregates these updates
+into a new global model.
+
+From this description, it might appear that individual local models are plainly
+visible to the coordinator. What if sensitive data could be extracted from them?
+Would this not be a violation of participants' data privacy?
+
+In fact, a key point about this process is that the updates are **not** sent in
+the plain! Rather, they are sent encrypted (or *masked*) so that the coordinator
+(and by extension, XayNet) learns almost nothing about the individual updates.
+Yet, it is nevertheless able to aggregate them in such a way that the resulting
+global model is unmasked.
+
+This is essentially what is meant by federated learning that is
+*privacy-preserving*, and is a key feature enabled by the PET protocol.
+
+## PET Protocol
+
+It is worth describing the protocol very briefly here, if only to better
+understand some of the configuration settings we will meet later. It is helpful
+to think of each round being divided up into several contiguous phases:
+
+**Start.** At the start of a round, the coordinator generates a collection of random *round
```

Comment:
not sure if we should deviate from the internal phase naming, **start** makes sense from a narrative point of view, but the connection to **idle** might not be directly obvious.

Reply:
I considered this too. But calling it Idle here, I felt might raise a question mark for the reader since the name is only really meaningful together with the context of the state machine, which is not mentioned in this guide. and as the other states aren't explicitly mentioned either, I felt just saying **Start** would be more intuitive here.

Reply:
makes sense.